### ISSUE TITLE
Callback takes up to 6 seconds to fire.

### ISSUE THREAD

**Issue by nicgutierrez** [react-native-community/react-native-image-picker#627]
### Description

Selecting an image from the gallery of about 6M will cause the picker to show but the callback will take up to 6 seconds to get fired.

### How to repeat issue and example

I'm not doing anything particular. Just following the example really, here's the options I'm using:
````
const options = {
    title: 'Select your avatar',
    storageOptions: {
        skipBackup: true,
        path: 'images'
    },
    mediaType: 'photo'
};
````

### Additional Information

* React Native version: 0.44.0
* Platform: iOS
* Development Operating System: MacOS
* Dev tools: iOS 9.x - Xcode 8.3.3 - iPhone 6

**Comment by Clowning** [comment]
Same issue for me!

**Comment by ajonno** [comment]
me too. is a showstopper for our app so unless can workaround will need to look at alternatives. if anyone has suggestion please let me know, thanks!

**Comment by nicgutierrez** [comment]
After trying out https://github.com/ivpusic/react-native-image-crop-picker, it seems that the issue might be related with the base64 data? The only reason why I'm saying this, is because react-native-image-crop-picker returns the image right away when base64 is switched off (default), however, if you turn it on, it takes a little bit longer.

Any ideas?

**Comment by eflath** [comment]
We are experiencing this same issue. Very long lag between selecting an image and getting the callback. My first thought was a threading issue (iOS) but after inspecting the code that doesn't seem likely.

We tried a couple things as workarounds though and did find something ...
I couldn't tell you why this worked, but we eliminated the lag just by setting maxWidth and maxHeight. Even setting them to implausibly large values (10,000) cut the lag down to acceptable levels.
